### PR TITLE
many: finish sharing a single TaskRunner with all the the managers

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
start of this is split out as #5473

File: overlord/ifacestate/ifacemgr.go [hunk 1]
```diff
@@ -61,43 +58,52 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, extraInterfaces
...
+	addHandler("gadget-connect", m.doGadgetConnect, nil)
 
 	// helper for ubuntu-core -> core
-	runner.AddHandler("transition-ubuntu-core", m.doTransitionUbuntuCore, m.undoTransitionUbuntuCore)
+	addHandler("transition-ubuntu-core", m.doTransitionUbuntuCore, m.undoTransitionUbuntuCore)
 
-	return m, nil
-}
+	// interface tasks might touch more than the immediate task target snap, serialize them
+	runner.AddBlocked(func(t *state.Task, running []*state.Task) bool {
```
Comment (stolowski) [inline overlord/ifacestate/ifacemgr.go]: Nice! :heart:

File: overlord/overlord_test.go [hunk 2]
```diff
@@ -218,6 +214,30 @@ func (ovs *overlordSuite) TestTrivialRunAndStop(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (ovs *overlordSuite) TestUnknownTasks(c *C) {
```
Comment (stolowski) [inline overlord/overlord_test.go]: :+1:

File: overlord/overlord.go [hunk 3]
```diff
@@ -105,10 +104,13 @@ func New() (*Overlord, error) {
 	o.stateEng = NewStateEngine(s)
 	o.runner = state.NewTaskRunner(s)
 
-	o.unknownMgr = NewUnknownTaskManager(s)
-	o.stateEng.AddManager(o.unknownMgr)
+	// any unknown task should be ignored and succeed
+	matchAnyUnknownTask := func(_ *state.Task) bool {
+		return true
+	}
+	o.runner.AddOptionalHandler(matchAnyUnknownTask, handleUnknownTask, nil)
```
Comment (stolowski) [inline overlord/overlord.go]: Nice.

**Pull request merged**